packages/mobile/src/fiatExchanges/FiatExchangeAmount.tsx

```diff
@@ -79,10 +78,7 @@ export function ExchangeTradeScreen({ navigation, route }: Props) {
   }
 
   function goNext() {
-    const now = Date.now()
-    const isLimitReached = isPaymentLimitReached(now, recentPayments, dollarAmount.toNumber())
-    if (isLimitReached) {
-      dispatch(showLimitReachedError(now, recentPayments, localExchangeRate, localCurrencySymbol))
+    if (isPaymentLimitReached(dollarAmount)) {
```

Comment:
Nit: I think this is much cleaner overall. Only thing is it's not intuitive to me that the error is being shown within `isPaymentLimitReached`. I'd suggest either changing the function name to try to capture that or breaking out the `showError` to be within the if block (i.e., `isPaymentLimitReached` would only return a boolean but not display an error either way)

Reply:
I agree it's not intuitive, the reason I wanted to leave it here is that the `showLimitReachedError` has a lot of dependencies that have to be imported in every place it's used just to show the error message and I though the usage would be much simpler if we could 'hide' those dependencies within this method. I'll try to think of a rename to the method to make clear what's happening.